### PR TITLE
Fix#123

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

**Comment by patrickelectric** [comment]
This is in WIP, does not represent the final version of this approach.
The VariantTree is what can be reviewed.

**Comment by patrickelectric** [comment]
This PR is close now to the final idea.

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -86,6 +98,33 @@ Ping::Ping() : Sensor()
     }
 }
 
+void Ping::checkPingConfigurationSettings()
+{
+    for(const auto& key : _pingConfiguration.keys()) {
+        _pingConfiguration[key].value = _pingConfiguration[key].defaultValue;
+    }
+
+    QVariant pingConfigurationVariant = SettingsManager::self()->getMapValue({"Ping", "PingConfiguration", "0"});
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: Maybe use device id for the path?

File: src/sensor/ping.h [hunk 2]
```diff
@@ -483,8 +485,71 @@ class Ping : public Sensor
     void firmwareUpdatePercentage();
     void flash(const QString& portLocation, const QString& firmwareFile, int baud = 57600, bool verify = true);
 
+    void checkPingConfigurationSettings();
```
Comment (jaxxzer) [inline src/sensor/ping.h]: I would reword these to load, save, and apply.
